vhdl_lang/src/syntax/subprogram.rs

```diff
@@ -106,11 +106,11 @@ pub fn parse_subprogram_instantiation(
     stream: &TokenStream,
     diagnostics: &mut dyn DiagnosticHandler,
 ) -> ParseResult<SubprogramInstantiation> {
-    let id = stream.get_token_id();
+    let start_token = stream.get_current_token_id();
```

Comment:
You mean replacing
```rust
let start_token = stream.get_current_token_id();
let tok = stream.peek_expect()?;
```
with some kind of 
```rust 
let tok = stream.peek_expect()?;
let start_token = stream.get_token_id(tok).unwrap();
```

I am not too sure if that's worth it. Since I would have to call `TokenStream::idx_of` we would need an extra division and some conditionals. Probably not a major issue, but I wanted to mention it 🤷 

If you insist, I will change it, though.

Reply:
You can leave it as it is.